serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc

```diff
@@ -1 +1,284 @@
+      },
+      {
+         "name":"shipping failed",
+         "code":"org.kie.kogito.ServiceException"
+      }
+   ]
+}
+----
+
+Error definition uses the FQCN of Java exceptions thrown by functions, in the given example, `org.kie.kogito.ServiceException` is thrown by they service calls defined as a  <<#custom-function, Java method>> here is the link:{kogito_sw_examples_url}/serverless-workflow-saga-quarkus/src/main/java/org/kie/kogito/PaymentService.java[source], however, they could be any type of link:{spec_doc_url}#workflow-functions[functions], i.e REST, OpenAPI, gRPC, etc.
+
+For more details about error handling see xref:core/understanding-workflow-error-handling.adoc[Understanding workflow error handling].
+
+[#custom-function]
+.Example custom function using a Java class and method
+[source,json]
+----
+{
+   "name":"reserveStock",
+   "type":"custom",
+   "operation":"service:org.kie.kogito.PaymentService::processPayment"
+}
+----
+
+Kogito workflow engine controls the execution flow keeping track of the steps to compensate where the compensation states are executed in a reverse order of each completed step.
+
+The engine is a stateful, which allows Saga to be long-lived containing wait states, such as callbacks. After each wait state, the workflow is persisted and can continue once it receives another request or event.
+
+For simplicity, the current example shows the Saga as a request-response, also called straight-through workflow, where in the same start request the workflow is totally executed and the response is returned.
+
+=== Running and testing the example
+
+==== Creating a new Success Order
+:order_saga_workflow_path: order_saga_error_workflow
+//POST /{order_saga_workflow_path}
+
+.Example success request
+[source,shell]
+----
+curl -L -X POST "http://localhost:8080/order_saga_error_workflow" -H 'Content-Type: application/json' --data-raw '{
+ "workflowdata": {
+  "orderId": "03e6cf79-3301-434b-b5e1-d6899b5639aa"
+ }
+}'
+----
+
+.Example success response
+[source,json]
+----
+{
+   "id":"b5c0bf16-1e37-4d7a-82cd-610809090d9c",
+   "workflowdata":{
+      "orderId":"03e6cf79-3301-434b-b5e1-d6899b5639aa",
+      "stockResponse":{
+         "type":"SUCCESS",
+         "resourceId":"dc32abe6-9706-4061-8e96-910d8e06728d"
+      },
+      "paymentResponse":{
+         "type":"SUCCESS",
+         "resourceId":"505259d9-1c12-40ea-af5d-679e2cd89394"
+      },
+      "shippingResponse":{
+         "type":"SUCCESS",
+         "resourceId":"d6e2d538-0229-4b8e-a363-17ebabdb3585"
+      },
+      "orderResponse":{
+         "type":"SUCCESS",
+         "resourceId":"03e6cf79-3301-434b-b5e1-d6899b5639aa"
+      }
+   }
+}
+----
+
+The response contains the workflow data with nested attributes representing the responses related to the execution of each step, either success or failure.
+The `orderResponse` indicates if the order can be confirmed by the client starting the Saga process, in case of `success` or canceled, in case of `error`.
```

Comment:
```suggestion
The `orderResponse` indicates if the order can be confirmed by the client starting the Saga process, in case of `success` or canceled, in case of `error`.
```
The end of the sentence is a bit confusing, can you clarify that?

Reply:
+1